### PR TITLE
SAML: Add idp signing configs

### PR DESCRIPTION
**Pull request by tung2744**
requires https://github.com/authgear/authgear-server/pull/4565
ref DEV-1719

### CODE REVIEW

**Comment by louischan-oursky** [comment]
Build failed.

**Comment by tung2744** [comment]
Fixed the test, thanks!

File: pkg/lib/config/secret_cert.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package config
+
+var _ = SecretConfigSchema.Add("X509CertPem", `
+{
+	"type": "string",
+	"format": "x_x509_cert_pem"
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_cert.go:34]: `x_x509_certificate_pem`

File: cmd/authgear/cmd/cmdinternal/saml.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+var cmdInternalSaml = &cobra.Command{
+	Use:   "saml",
+	Short: "SAML commands",
+}
+
+var cmdInternalSamlGenerateSigningKey = &cobra.Command{
+	Use:   "generate-signing-key { common-name }",
+	Short: "Generate a signing key with a x.509 cert",
```
Comment (louischan-oursky) [inline cmd/authgear/cmd/cmdinternal/saml.go:31]: `X.509 certificate` The `X` is in uppercase.

File: pkg/lib/config/saml.go [hunk 3]
```diff
@@ -40,3 +42,55 @@ func (c *SAMLConfig) ResolveProvider(id string) (*SAMLServiceProviderConfig, boo
...
+
+type SAMLSigningSignatureMethod string
+
+const (
+	SAMLSigningSignatureMethodRSASHA256 SAMLSigningSignatureMethod = "RSAwithSHA256"
+)
+
+var _ = Schema.Add("SAMLSigningDigestMethod", `
+{
+	"enum": ["SHA256", "SHA1"]
```
Comment (louischan-oursky) [inline pkg/lib/config/saml.go:87]: Support SHA256 only. Remove the support for SHA1

**Comment by louischan-oursky** [comment]
OK.

File: pkg/lib/config/saml.go (hunk 3 above)
Comment (tung2744) [reply pkg/lib/config/saml.go:87]: It seems SHA1 is still widely supported as digest method in different implementations, but I think we can add it back easily if necessary so remove first.

**Comment by tung2744** [comment]
Updated, thanks!

File: pkg/lib/config/secret_cert.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package config
+
+import (
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+)
+
+var _ = SecretConfigSchema.Add("X509Cert", `
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_cert.go:9]: X509Certificate

File: pkg/lib/config/secret_cert.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package config
+
+var _ = SecretConfigSchema.Add("X509CertPem", `
```
[the remark this answers is not in the document]
Comment (louischan-oursky) [reply pkg/lib/config/secret_cert.go:31]: X509CertificatePEM

File: pkg/lib/config/secret_saml.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
+package config
+
+var _ = SecretConfigSchema.Add("SAMLIdpSigningMaterials", `
+{
+	"type": "object",
+	"properties": {
+		"certificates": {
+			"type": "array",
+			"items": { "$ref": "#/$defs/SAMLIdpSigningCert" },
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_saml.go:9]: SAMLIdpSigningCertificate

File: pkg/lib/config/secret_saml.go [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+	return nil, false
+}
+
+var _ SecretItemData = &SAMLIdpSigningMaterials{}
+
+func (s *SAMLIdpSigningMaterials) SensitiveStrings() []string {
+	return nil
+}
+
+var _ = SecretConfigSchema.Add("SAMLIdpSigningCert", `
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_saml.go:37]: SAMLIdpSigningCertificate

File: pkg/lib/config/secret_saml.go [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+{
+	"type": "object",
+	"properties": {
+		"certificate": { "$ref": "#/$defs/X509Cert" },
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_saml.go:41]: X509Certificate

File: pkg/lib/config/secret_saml.go [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+type SAMLIdpSigningCert struct {
+	Certificate *X509Cert `json:"certificate,omitempty"`
+	Key         *JWK      `json:"key,omitempty"`
+}
+
+var _ = SecretConfigSchema.Add("SAMLSpSigningMaterials", `
+{
+	"type": "array",
+	"items": { "$ref": "#/$defs/SAMLSpSigningCert" }
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_saml.go:56]: SAMLSpSigningCertificate

File: pkg/lib/config/secret_saml.go [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+type SAMLSpSigningMaterials []SAMLSpSigningCert
+
+var _ SecretItemData = &SAMLSpSigningMaterials{}
+
+func (s *SAMLSpSigningMaterials) SensitiveStrings() []string {
+	return nil
+}
+
+var _ = SecretConfigSchema.Add("SAMLSpSigningCert", `
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_saml.go:68]: SAMLSpSigningCertificate

File: pkg/lib/config/secret_saml.go [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+{
+	"type": "object",
+	"properties": {
+		"service_provider_id": { "type": "string" },
+		"certificates": {
+			"type": "array",
+			"items": { "$ref": "#/$defs/X509Cert" }
```
Comment (louischan-oursky) [inline pkg/lib/config/secret_saml.go:75]: X509Certificate

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by tung2744** [comment]
Fixed, thanks!

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**